src/stats.rs

```diff
@@ -107,8 +107,19 @@ impl Collector {
             loop {
                 interval.tick().await;
 
-                for stats in SERVER_STATS.read().values() {
-                    stats.address_stats().update_averages();
+                // Hold read lock for duration of update to retain all server stats
```

Comment:
I am confused here. We only have one instance of the collector running, so there is only one task updating averages. How can we run `update_averages()` twice or more even with multiple server connections connected to the same address?

Reply:
This issue here comes from all the servers having the same reference to the AddressStats. When we update address stat's old_total. The next server that we try to update averages for will see no difference between the current total and the old total cause it was updated in the previous iteration of the loop

Reply:
I have an example of this in the description of the PR

Reply:
Aaah okay. Confusing! Arcs are tricky. Ok, sounds good, I think this is fine for now. Would be good to investigate further and maybe remove the leaked reference?